plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeUpdatablePageSource.java

```diff
@@ -393,15 +395,16 @@ public void updateRows(Page page, List<Integer> columnValueAndRowIdChannels)
         int unmodifiedColumnIndex = 0;
         int updatedColumnIndex = 0;
         Block[] fullPage = new Block[allDataColumns.size()];
+        ColumnarRow columnarRow = toColumnarRow(rowIdBlock);
 
         for (int targetChannel = 0; targetChannel < allDataColumns.size(); targetChannel++) {
             if (updatedColumns.contains(allDataColumns.get(targetChannel))) {
                 fullPage[targetChannel] = page.getBlock(columnChannelMapping.get(updatedColumnIndex));
                 updatedColumnIndex++;
             }
             else {
-                Block unmodifiedColumns = rowIdBlock.getChildren().get(1);
-                fullPage[targetChannel] = unmodifiedColumns.getChildren().get(unmodifiedColumnIndex);
+                Block unmodifiedColumns = columnarRow.getField(1);
+                fullPage[targetChannel] = toColumnarRow(unmodifiedColumns).getField(unmodifiedColumnIndex);
```

Comment:
You can move this one out as well

Reply:
I actually tried that and it made some tests fail with IndexOutOfBound in `Block unmodifiedColumns = columnarRow.getField(1);`

Reply:
Right, that might happen if all rows are being updated. You could make the field Optional or nullable. Might be worth it if `toColumnarRow` might do a meaningful amount of work